src/utils/queryKeyFactory.js

```diff
@@ -0,0 +1,112 @@
+import { createQueryKeys, mergeQueryKeys } from '@lukemorales/query-key-factory';
+import { fetchCourseMetadata } from '../components/app/routes/queries/courseMetadata';
+import { fetchUserEntitlements } from '../components/app/routes/queries/userEntitlements';
+import { fetchCanRedeem } from '../components/app/routes/queries/canRedeemCourse';
+import { fetchEnterpriseLearnerData } from '../components/app/routes/queries/enterpriseLearner';
+import { fetchEnterpriseCourseEnrollments } from '../components/app/routes/queries/enterpriseCourseEnrollments';
+import {
+  fetchBrowseAndRequestConfiguration,
+  fetchCouponCodeRequests,
+  fetchLicenseRequests,
+} from '../components/app/routes/queries/subsidies/browseAndRequest';
+import { fetchRedeemablePolicies } from '../components/app/routes/queries/subsidies/policies';
+import { fetchSubscriptions } from '../components/app/routes/queries/subsidies/subscriptions';
+import { fetchCouponCodes } from '../components/app/routes/queries/subsidies/couponCodes';
+import { fetchEnterpriseOffers } from '../components/app/routes/queries/subsidies/enterpriseOffers';
+import { fetchEnterpriseCuration } from '../components/app/routes/queries/contentHighlights';
+import { SUBSIDY_REQUEST_STATE } from '../components/enterprise-subsidy-requests';
+
+export const enterprise = createQueryKeys('enterprise', {
+  enterpriseCustomer: (enterpriseUuid) => ({
+    queryKey: [enterpriseUuid],
+    contextQueries: {
+      contentHighlights: {
+        queryKey: null,
+        contextQueries: {
+          configuration: {
+            queryKey: null,
+            queryFn: async ({ queryKey }) => fetchEnterpriseCuration(queryKey[2]),
+          },
+        },
+      },
+      course: {
+        queryKey: null,
+        contextQueries: {
+          contentMetadata: (courseKey) => ({
+            queryKey: [courseKey],
+            queryFn: async ({ queryKey }) => fetchCourseMetadata(queryKey[2], courseKey),
+          }),
+          canRedeem: (availableCourseRunKeys) => ({
+            queryKey: [availableCourseRunKeys],
+            queryFn: async ({ queryKey }) => fetchCanRedeem(queryKey[2], availableCourseRunKeys),
+          }),
+        },
+      },
+      enrollments: {
+        queryKey: null,
+        queryFn: async ({ queryKey }) => fetchEnterpriseCourseEnrollments(queryKey[2]),
+      },
+      subsidies: {
+        queryKey: null,
+        contextQueries: {
+          browseAndRequest: (userEmail) => ({
+            queryKey: [userEmail],
+            contextQueries: {
+              configuration: {
+                queryKey: null,
+                queryFn: async ({ queryKey }) => fetchBrowseAndRequestConfiguration(queryKey[2], queryKey[4]),
+              },
+              endpoints: (state = SUBSIDY_REQUEST_STATE.REQUESTED) => ({
```

Comment:
[nit/clarification] Is this `endpoints` grouping here as a way to DRY up the `state` argument relevant to both `licenseRequests` and `couponCodeRequests`? If so, I think it's reasonable to have, however, perhaps renaming this grouping key to `requests`? All these queries are associated endpoints at the end of the day, so the `endpoints` key here isn't adding much semantically.